### PR TITLE
feat(vm): ability to choose the cores per socket when creating or editing a VM

### PR DESCRIPTION
**Pull request by badrAZ**
Fixes vatesfr/xo-web#130

### CODE REVIEW

File: src/xapi-object-to-xo.js [hunk 1]
```diff
@@ -79,7 +79,11 @@ const TRANSFORMS = {
       tags: obj.tags,
       name_description: obj.name_description,
       name_label: obj.name_label || obj.$master.name_label,
-      xosanPackInstallationTime: toTimestamp(obj.other_config.xosan_pack_installation_time)
+      xosanPackInstallationTime: toTimestamp(obj.other_config.xosan_pack_installation_time),
+      cpus: {
+        cores: +obj.cpu_info.cpu_count,
+        sockets: +obj.cpu_info.socket_count
```
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Are we sure `obj.cpu_info.{cpu,socket}_count` are always defined?

On XS 6.2, 6.5, 7.0, 7.1?

File: src/xapi-object-to-xo.js [hunk 2]
```diff
@@ -367,6 +371,8 @@ const TRANSFORMS = {
       } = obj.platform)
     }
 
+    vm.coresPerSocket = obj.platform['cores-per-socket']
```
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Shouldn't it be a number?

File: src/xapi/mixins/vm.js [hunk 3]
```diff
@@ -259,6 +259,12 @@ export default {
       }
     },
 
+    coresPerSocket: {
+      set (core, vm) {
```
Comment (julien-f) [inline src/xapi/mixins/vm.js]: `core` → `coresPerSocket` or `n`.

File: src/xapi-object-to-xo.js (hunk 1 above)
Comment (badrAZ) [inline src/xapi-object-to-xo.js]: It is only defined in 7.0, 7.1,  i put 4 cores as default value?

File: src/xapi-object-to-xo.js (hunk 2 above)
Comment (badrAZ) [inline src/xapi-object-to-xo.js]: Yes, it should. I will cast it.

File: src/xapi-object-to-xo.js (hunk 1 above)
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Does it makes sense to have default values?

If not, simply set this values to `undefined`.

File: src/xapi-object-to-xo.js (hunk 2 above)
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Be careful to not cast `undefined`.

File: src/xapi-object-to-xo.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ const TRANSFORMS = {
       name_description: obj.name_description,
       name_label: obj.name_label || obj.$master.name_label,
       xosanPackInstallationTime: toTimestamp(obj.other_config.xosan_pack_installation_time),
-      cpus: {
+      cpus: obj.cpu_info && {
```
Comment (julien-f) [inline src/xapi-object-to-xo.js]: I would prefer avoiding conditional nested props.

File: src/xapi-object-to-xo.js [hunk 5]
```diff
@@ -371,7 +371,7 @@ const TRANSFORMS = {
       } = obj.platform)
     }
 
-    vm.coresPerSocket = obj.platform['cores-per-socket']
+    vm.coresPerSocket = obj.platform['cores-per-socket'] && +obj.platform['cores-per-socket']
```
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Use an intermediary variable.

File: src/xapi-object-to-xo.js (hunk 4 above)
Comment (julien-f) [inline src/xapi-object-to-xo.js]: Use an intermediary variable.

**Pull request merged**